### ISSUE TITLE
`xm.save()` should not trigger buffer aliasing

### ISSUE THREAD

**Issue by mcuiaws** [pytorch/xla#8422]
## 🐛 Bug

When `xm.save()` is called on a subset of tensors, it calls down to `_xla_sync_multi(sync_xla_data=True)` which eventually calls down to `XLAGraphExecutor::SyncTensorsGraph()` with `sync_ltc_data=true` as if it's doing a step marker. This causes buffer aliasing to be performed by `XLAGraphExecutor::Compile()`. However, according to the detailed comments in [xla_graph_executor.cpp#L1336](https://github.com/pytorch/xla/blob/v2.5.1/torch_xla/csrc/xla_graph_executor.cpp#L1336), buffer aliasing should only be performed at a step marker when *all live tensors* are being sync'd. This is obviously not the case for `xm.save()`.

As a result, the `xm.mark_step()` following `xm.save()` will have parameters which refer to donated buffers which are deleted by PJRT, triggering [this](https://github.com/pytorch/xla/blob/v2.5.1/torch_xla/csrc/runtime/pjrt_computation_client.h#L192) `XLA_CHECK()` failure in `PjRtData::GetHandle()`.

## To Reproduce

A small example to reproduce.

```
import os
import torch
import torch_xla
import torch_xla.core.xla_model as xm

def main():
    xla_device = xm.xla_device()
    t0 = torch.tensor([1], device=xla_device)
    t1 = torch.tensor([2], device=xla_device)
    xm.mark_step()

    t2 = t0 + t1
    t1.add_(1)

    # Save the new value of t1 should not result in the old value
    # being donated...
    xm.save(t1, os.devnull)

    # otherwise this mark_step could crash, or compute the wrong value for t2.
    xm.mark_step()
    assert t2.item() == 3

if __name__ == "__main__":
    main()
```

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]: neuron trn1
 - torch_xla version: 2.5.1

**Comment by mcuiaws** [comment]
Our PJRT device plugin will mark a PJRT_Buffer as deleted if it's donated to an output buffer, causing the above mentioned XLA_CHECK() to fail. In case other device plugins do not have this behavior, here is a [patch](https://github.com/mcuiaws/pytorch-xla/commit/65ab5787932ffaf99829e829456bc0a007163b50) that forces a crash in xla_graph_executor.cpp.

**Comment by miladm** [comment]
@qihqi can you please review this issue?

**Comment by jeffhataws** [comment]
Let's get a PR for the fix in before branch-cut for 2.6, which is today I believe.
